tests/unittests/tests-libc/tests-libc.c

```diff
@@ -30,6 +30,38 @@ static void test_libc_strscpy(void)
     TEST_ASSERT_EQUAL_INT(strscpy(buffer, "empty", 0), -E2BIG);
 }
 
+static void test_libc_swprintf(void)
+{
+    string_writer_t sw;
+    char buffer[32];
+    int res;
+
+    string_writer_init(&sw, buffer, sizeof(buffer));
+
+    /* check that string can be written completely */
+    res = swprintf(&sw, "Hello World!");
+    TEST_ASSERT_EQUAL_INT(res, 12);
+    TEST_ASSERT_EQUAL_INT(string_writer_len(&sw), 12);
+    TEST_ASSERT_EQUAL_INT(strcmp("Hello World!", string_writer_str(&sw)), 0);
+
+    /* check that we can add to the string */
+    res = swprintf(&sw, "0123456789");
+    TEST_ASSERT_EQUAL_INT(res, 10);
```

Comment:
```suggestion
    /* Writing 10 characters, returns 10 bytes written */
    res = swprintf(&sw, "0123456789");
    TEST_ASSERT_EQUAL_INT(res, 10);
```

(This immediately clarifies to the reader: Bytes written is returned and not "bytes in buffer remaining". Compares nicely to the first test case)